### PR TITLE
Handle domains on inventory

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !27212

### CODE REVIEW

**Comment by stonebuzz** [comment]
When changing the domain, the old one should no longer be linked to the computer  (IMHO)

image

- Import an inventory file
- change domain 
- re-import the inventory file

We should have only one domain

**Comment by trasher** [comment]
Thinking about that, it may cause issues... Domains_Item does not have is_dynamic, therefore we cannopt known if a relation has been manually added (in that case, we must keep it).
The only way to get that fixed properly is certainly to add `is_dynamic` on Domain_Items

**Review by cedric-anne — APPROVED** [review]

File: tests/functionnal/Glpi/Inventory/Assets/Computer.php [hunk 1]
```diff
@@ -1311,4 +1311,96 @@ public function testTransferNotAllowedForSoftware()
...
+        //check domain has been created
+        $first_id = $domain->fields['id'];
+        $domain = new \Domain();
+        $this->boolean($domain->getFromDBByCrit(['name' => addslashes("workgroup'ed another time")]))->isTrue();
+
+        //check relation has been created - and there is only one remaining
+        $domain_item = new \Domain_Item();
+        $this->boolean($domain_item->getFromDBByCrit(['domains_id' => $domain->fields['id']]))->isTrue();
+        $this->boolean($domain_item->getFromDBByCrit(['domains_id' => $first_id]))->isTrue();
+        $this->boolean($domain_item->getFromDBByCrit(['domains_id' => $first_id, 'is_deleted' => 0]))->isFalse();
```
Comment (cedric-anne) [inline tests/functionnal/Glpi/Inventory/Assets/Computer.php:1369]: Probably easier to read.
```suggestion
        $this->boolean($domain_item->getFromDBByCrit(['domains_id' => $first_id, 'is_deleted' => 1]))->isTrue();
```

Comment (cedric-anne) [inline tests/functionnal/Glpi/Inventory/Assets/Computer.php:1369]: `'itemtype' => 'Computer', 'items_id' => $computers_id` may be added to requests to ensure we are checking relation to inventoried item.

File: tests/functionnal/Glpi/Inventory/Assets/Computer.php [hunk 2]
```diff
@@ -1311,4 +1311,96 @@ public function testTransferNotAllowedForSoftware()
...
+
+        //check domain has been created
+        $first_id = $domain->fields['id'];
+        $domain = new \Domain();
+        $this->boolean($domain->getFromDBByCrit(['name' => addslashes("workgroup'ed another time")]))->isTrue();
+
+        //check relation is still present - and non dynamic one as well
+        $domain_item = new \Domain_Item();
+        $this->boolean($domain_item->getFromDBByCrit(['domains_id' => $domain->fields['id']]))->isTrue();
+        $this->boolean($domain_item->getFromDBByCrit(['domains_id' => $ndyn_domain->fields['id']]))->isTrue();
```
Comment (cedric-anne) [inline tests/functionnal/Glpi/Inventory/Assets/Computer.php:1404]: `'itemtype' => 'Computer', 'items_id' => $computers_id` may be added to requests to ensure we are checking relation to inventoried item.

**Review by cedric-anne — APPROVED** [review]

File: tests/functionnal/Glpi/Inventory/Assets/Computer.php (hunk 1 above)
Comment (trasher) [reply tests/functionnal/Glpi/Inventory/Assets/Computer.php:1369]: This is not necessary, we're working on domains added locally, and I'm really checking domain relation, whatever the item it is linked to (it's tested in the 1st case)

**Review by cedric-anne — APPROVED** [review]

**Comment by trasher** [comment]
@stonebuzz this one should be OK now

**Review by stonebuzz — APPROVED** [review]

**Pull request merged**